Lib/ftplib.py

```diff
@@ -324,6 +331,8 @@ def makeport(self):
         sock.listen(1)
         port = sock.getsockname()[1] # Get proper port
         host = self.sock.getsockname()[0] # Get proper host
+        if self.externalip and socket.inet_aton(self.externalip):
```

Comment:
This breaks IPv6 support. You have to use ``inet_pton`` if available. Or rather default ``externalip`` to ``None`` and check for None.

Reply:
good catch, thanks for your review